frontend/src/components/dialogs/SecretDialogGeneric.vue

```diff
@@ -92,33 +92,35 @@ export default {
       const validators = {
         data: {
           required,
-          isYAML: () => isObject(this.secretYAML)
+          isYAML: () => isObject(this.parsedYAML)
         }
       }
       return validators
     },
     isCreateMode () {
       return !this.secret
     },
-    secretYAML () {
-      try {
-        return yaml.load(this.data)
-      } catch (e) {
-        return undefined
-      }
-    },
     secretData () {
-      return isObject(this.secretYAML) ? this.secretYAML : {}
+      return isObject(this.parsedYAML) ? this.parsedYAML : {}
     }
   },
   methods: {
     onInput (value) {
       this.$emit('input', value)
     },
+    async onChangeData () {
+      try {
+        this.parsedYAML = await this.$yaml.load(this.data)
+      } catch (err) {
+        this.parsedYAML = undefined
+      }
+      this.$v.data.$touch()
+    },
```

Comment:
```suggestion
    async parseYAML () {
      try {
        this.parsedYAML = await this.$yaml.load(this.data)
      } finally {
        if (!isObject(this.parsedYAML)) {
          this.parsedYAML = {}
        }
      }
    },
    onChange () {
      this.parseYAML()
      this.$v.data.$touch()
    },
```

Reply:
I used parts of your proposal. However, we still need to reset to `undefined` on error otherwise the validation will not work.